packages/react/src/components/Table/StatefulTable.test.jsx

```diff
@@ -74,40 +71,32 @@ describe('stateful table with real reducer', () => {
         actions={mockActions}
       />
     );
-
     expect(statefulTable.text()).not.toContain('myButtons');
+    await act(async () => {
+      await statefulTable.find(RowActionsCell).first().props().onApplyRowAction('edit', 'row-1');
+    });
 
-    return statefulTable
-      .find(RowActionsCell)
-      .first()
-      .props()
-      .onApplyRowAction('edit', 'row-1')
-      .then(() => {
-        expect(true).toBeTruthy();
-        expect(statefulTable.text()).toContain('myButtons');
-      });
+    expect(true).toBeTruthy();
+    expect(statefulTable.text()).toContain('myButtons');
   });
-
-  it('render nestedRows', () => {
+  it('render nestedRows', async () => {
     const tableId = 'tableId';
     render(<StatefulTableWithNestedRowItems id={tableId} actions={mockActions} />);
-
     expect(screen.queryByText('whiteboard can eat 2A')).toBeNull();
-
-    fireEvent.click(
+    userEvent.click(
       screen
         .getByText('whiteboard can eat 2')
         .closest('tr')
         .querySelector('.bx--table-expand__button')
     );
-
     expect(screen.getByText('whiteboard can eat 2A')).toBeTruthy();
-
-    fireEvent.click(screen.getByTestId(`${tableId}-row-2_A-row-actions-cell-overflow`));
-    fireEvent.click(screen.getByText('Add'));
+    userEvent.click(screen.getByTestId(`${tableId}-row-2_A-row-actions-cell-overflow`));
+    await act(async () => {
+      const addBtn = await screen.findByText('Add');
+      userEvent.click(addBtn);
+    });
```

Comment:
This needs to be wrapped in act because of the async nature of the dispatch used in the stateful table, or we get an error about state changes outside of act in the jest tests.